src/freenet/node/NodeDispatcher.java

```diff
@@ -234,6 +234,8 @@ public boolean handleMessage(Message m) {
 				rejectRequest(m, node.nodeStats.sskInsertCtr);
 			} else if(spec == DMT.FNPGetOfferedKey) {
 				rejectRequest(m, node.failureTable.senderCounter);
+			} else {
+				return false;
 			}
 			return true;
```

Comment:
No. We want to consume the message if we reject it, but not if it's unrecognised.